### PR TITLE
Add Dual Kawase Blur

### PR DESCRIPTION
**Pull request by Lyzev**
## Type of change

- [ ] Bug fix
- [x] New feature

## Description

I decided to implement Dual Kawase Blur because it performs better than the current blur method. I made some changes to important classes like Framebuffer, Blur and others to make it work. I did this to make sure the new blur algorithm fits well with the existing code and doesn't cause any major issues. We also took care to keep the code easy to maintain and make sure it doesn't affect the overall structure too much.

## Performance Overview

The graphs show that Dual Kawase Blur performs the best among all the blurs approximating Gaussian. Note that Meteor Client uses the Box Blur method.
image
image
[Source](https://discuss.cocos2d-x.org/t/implementation-and-application-of-high-performance-blur-dual-blur-on-2d-sprites/55819)

## Related issues

None.

# How Has This Been Tested?

**OS:** Garuda Linux x86_64
**Kernel:** 6.3.2-zen1-1-zen
image

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have added comments to my code in more complex areas.
- [x] I have tested the code in both development and production environments.

### CODE REVIEW

File: src/main/java/meteordevelopment/meteorclient/systems/modules/render/Blur.java [hunk 1]
```diff
@@ -9,86 +9,62 @@
...
-    private final Setting<Boolean> other = sgScreens.add(new BoolSetting.Builder()
-        .name("other")
-        .description("Applies blur to all other screen types.")
-        .defaultValue(true)
-        .build()
-    );
-
-    private Shader shader;
-    private Framebuffer fbo1, fbo2;
+    private final Setting<Boolean> meteor = sgScreens.add(new BoolSetting.Builder().name("meteor").description("Applies blur to Meteor screens.").defaultValue(true).build());
```
Comment (GreemDev) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/render/Blur.java:58]: why did you remove the newlines for settings, every single other module has it this way and it is way more readable

Comment (Lyzev) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/render/Blur.java:58]: The reformatting feature of IntelliJ was used. My mistake, I will correct it.

Comment (Lyzev) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/render/Blur.java]: Done.

Comment (GreemDev) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/render/Blur.java]: > The reformatting feature of IntelliJ was used. My mistake, I will correct it.

I was hoping it was something like that and not a weird conscious decision, good job 👍🏼

File: src/main/java/meteordevelopment/meteorclient/systems/modules/render/Blur.java [hunk 2]
```diff
@@ -9,42 +9,53 @@
...
-        Fancy,
-        Fast
-    }
-
     private final SettingGroup sgGeneral = settings.getDefaultGroup();
     private final SettingGroup sgScreens = settings.createGroup("Screens");
 
-    // General
+    // Strength-Levels from https://github.com/jonaburg/picom/blob/a8445684fe18946604848efb73ace9457b29bf80/src/backend/backend_common.c#L372
+    private final Pair<Integer, Double>[] strengths = new Pair[]{
```
Comment (RacoonDog) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/render/Blur.java:28]: Use fastutil primitive pairs (`IntDoubleImmutablePair`)

**Review by RacoonDog — CHANGES_REQUESTED** [review]
looks interesting

File: src/main/java/meteordevelopment/meteorclient/systems/modules/render/Blur.java (hunk 2 above)
Comment (Lyzev) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/render/Blur.java:28]: I did not know that this class exists. I will adapt my code as soon as possible.

Comment (Lyzev) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/render/Blur.java]: Done. If you have any other suggestions, let me know!
